build-tools/packages/build-cli/src/commands/generate/changeset.ts

```diff
@@ -119,11 +124,32 @@ export default class GenerateChangesetCommand extends BaseCommand<typeof Generat
 		const remote = await repo.getRemote(context.originRemotePartialUrl);
 
 		if (remote === undefined) {
-			// Logs and exits
 			this.error(`Can't find a remote with ${context.originRemotePartialUrl}`, { exit: 1 });
 		}
 		this.log(`Remote for ${context.originRemotePartialUrl} is: ${chalk.bold(remote)}`);
 
+		// If the branch flag was passed explicitly, we don't want to prompt the user to select one. We can't check for
+		// undefined because there's a default value for the flag.
+		const usedBranchFlag = this.argv.includes("--branch") || this.argv.includes("-b");
+		if (!usedBranchFlag) {
+			const { packages } = await repo.getChangedSinceRef(branch, remote, context);
+
+			if (packages.length > BRANCH_PROMPT_LIMIT) {
+				const answer = await prompts({
+					type: "select",
+					name: "selectedBranch",
+					message: `More than ${BRANCH_PROMPT_LIMIT} packages were edited compared to ${branch}. Maybe you meant to select a different target branch?`,
```

Comment:
Nit. Striving for zero ambiguity for people who encounter the messages for the first time.
```suggestion
					message: `More than ${BRANCH_PROMPT_LIMIT} packages were edited compared to branch '${branch}'. Maybe you meant to select a different target branch?`,
```